### ISSUE TITLE
Flashing UI components when posting / sending a message

### ISSUE THREAD

**Issue by patch-e** [Squeegy/latest-chatty-2#72]
See: http://www.shacknews.com/chatty?id=29622660#item_29622660

tl;dr the way posts and messages were being sent through the API caused UIKit to be called on background threads. Not a good thing, but it finally became an issue in iOS 6. 

Some console errors about UIKit usage on background threads were occurring when posting/SM'ing and sometimes you could get the status bar, navigation bar, and half the screen to start flashing in a weird way. You could get it to happen almost everytime if you tried to post with < 5 chars, get the error alert, and then post again with > 5 chars. The flashing would begin after that.

I basically took the existing post/SM create functions and wrapped them in grand central dispatch async calls. Everything works fine after that.
